### PR TITLE
Add AmneziaWG protocol improvements

### PR DESCRIPTION
**Pull request by leninalive**

### CODE REVIEW

**Review by tiaga — APPROVED** [review]

File: src/Kconfig [hunk 1]
```diff
@@ -1,5 +1,5 @@
-config WIREGUARD
-	tristate "IP: WireGuard secure network tunnel"
+config AMNEZIAWG
+	tristate "IP: WireGuard (Amnezia VPN) secure network tunnel"
```
Comment (pokamest) [inline src/Kconfig:2]: "IP: AmneziaWG secure network tunnel"

File: src/main.c [hunk 2]
```diff
@@ -50,7 +50,7 @@ static int __init wg_mod_init(void)
 	if (ret < 0)
 		goto err_netlink;
 
-	pr_info("WireGuard " WIREGUARD_VERSION " loaded. See www.wireguard.com for information.\n");
+	pr_info("WireGuard " WIREGUARD_VERSION " (Amnezia VPN) loaded. See www.wireguard.com for information.\n");
```
Comment (pokamest) [inline src/main.c:53]: AmneziaWG" WIREGUARD_VERSION "loaded

File: src/main.c [hunk 3]
```diff
@@ -76,7 +76,7 @@ static void __exit wg_mod_exit(void)
 module_init(wg_mod_init);
 module_exit(wg_mod_exit);
 MODULE_LICENSE("GPL v2");
-MODULE_DESCRIPTION("WireGuard secure network tunnel");
+MODULE_DESCRIPTION("WireGuard (Amnezia VPN) secure network tunnel");
```
Comment (pokamest) [inline src/main.c:79]: AmneziaWG secure network tunnel

File: src/patches/000-initial-amneziawg.patch [hunk 4]
```diff
@@ -0,0 +1,833 @@
...
++#include "uapi/wireguard.h"
+ #include "crypto/zinc.h"
+ 
+ #include <linux/init.h>
+@@ -52,7 +50,7 @@ static int __init wg_mod_init(void)
+ 	if (ret < 0)
+ 		goto err_netlink;
+ 
+-	pr_info("WireGuard " WIREGUARD_VERSION " loaded. See www.wireguard.com for information.\n");
++	pr_info("WireGuard " WIREGUARD_VERSION " (Amnezia VPN) loaded. See www.wireguard.com for information.\n");
```
Comment (pokamest) [inline src/patches/000-initial-amneziawg.patch:228]: AmneziaWG WIREGUARD_VERSION loaded

File: src/patches/000-initial-amneziawg.patch [hunk 5]
```diff
@@ -0,0 +1,833 @@
...
+ 	pr_info("Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.\n");
+ 
+ 	return 0;
+@@ -78,7 +76,7 @@ static void __exit wg_mod_exit(void)
+ module_init(wg_mod_init);
+ module_exit(wg_mod_exit);
+ MODULE_LICENSE("GPL v2");
+-MODULE_DESCRIPTION("WireGuard secure network tunnel");
++MODULE_DESCRIPTION("WireGuard (Amnezia VPN) secure network tunnel");
```
Comment (pokamest) [inline src/patches/000-initial-amneziawg.patch:237]: AmneziaWG secure network tunnel

**Review by pokamest — CHANGES_REQUESTED** [review]

File: src/main.c (hunk 2 above)
Comment (pokamest) [reply src/main.c:53]: pr_info("AmneziaWG" WIREGUARD_VERSION " loaded. See amnezia.org for information.\n");

**Review by tiaga — APPROVED** [review]

**Review by pokamest — APPROVED** [review]

**Comment by korzhyc** [comment]
This won't build on multi-core system or without `-j 1` param.
https://github.com/amnezia-vpn/amneziawg-linux-kernel-module/blob/40a9e512f062a7e2868866109d2431c3fd321c4c/src/Makefile#L18-L19

**Pull request merged**